src/operator.py

```diff
@@ -0,0 +1,375 @@
+    drone = drones.get(request["id"])
+    if not drone:
+        response["success"] = False
+        response["message"] = "Invalid drone id"
+        return False
+    if control_task == "start_mission":
+        print("Executing start_mission...")
+        response = drone.start_mission()
+    elif control_task == "pause_mission":
+        print("Executing pause_mission...")
+        response = drone.pause_mission()
+    elif control_task == "resume_mission":
+        print("Executing resume_mission...")
+        response = drone.resume_mission()
+    elif control_task == "land_drone":
+        print("Executing land_drone...")
+        response = drone.land_drone()
+    elif control_task == "fly_home":
+        print("Executing fly_home...")
+        response = drone.fly_home()
+    else:
+        response["success"] = False
+        response["message"] = "Invalid control task"
+        return False
+    print("Control_drone service finished!")
+    return True
+
+
+@custom_service
+def query_topics(request, response):
+    print("Calling query_topics service...")
+    id = request["id"]
+    response["id"] = id
+    all_topics_response = []
+    if id == 0:
+        for k,v in all_topics.items():
+            all_topics_response.append({"name": k, "type": v})
+    else:
+        if id in drones:
+            all_topics_response = drones[id].topics
+            for sensor_id in drones[id].sensors:
+                all_topics_response += sensors[sensor_id].topics
+        elif id in sensors:
+            all_topics_response = sensors[id].topics
+        else:
+            response["success"] = False
+            response["message"] = "No drone or sensor with that id."
+            return False
+
+    response["all_topics"] = all_topics_response
+    response["success"] = True
+    response["message"] = "Successfully queried topics."
+    print(all_topics_response)
+    print("Query_topics service finished!")
+    return True
+
+############################
+# Drone -> Server Handlers #
+############################
+@custom_service
+def register_drone(request, response):
+    '''
+    :param request: dict of {drone_name: string, drone_type: string}
+    '''
+    print("Calling register_drone service...")
+    drone_name = request["drone_name"]
+    drone_type = request["drone_type"]
+    # Create new drone instance using base class constructor, which should then
+    # call child constructor corresponding to the drone_type
+    d=Drone.create(drone_name, drone_type)
+    successful=False
+
+    if d:
+        id = get_id(Drone)
+        print(f"Adding drone {id} to global drones map...")
+        d.id=id
+        drones[id] = d
+        drone_names[drone_name] = id
+        successful = True
+        response["success"] = successful
+        response["drone_id"] = id
+    if successful:
+        response["message"] = "Drone registered"
+    else:
+        response["message"] = "Failed to register drone"
```

Comment:
Since successful is only set True if the code goes through the "if d" part of the code, could we delete the successful variable, add the "if successful" body to the "if d" body, and make the "else" statement correspond to "if d" instead of "if successful"?